### PR TITLE
target/sam3x: saner uninitialized variable prevention

### PR DESCRIPTION
**Pull request by perigoso**
in #1076 we initialized variables to 0, but in the event that a new driver is added and not properly handled, we would try to do GPNVM bit operations on a flash peripheral with address 0x0, I think this is a saner approach, where if a driver is not known the command immediately fails

Signed-off-by: Rafael Silva <perigoso@riseup.net>

### CODE REVIEW

**Review by dragonmux — APPROVED** [review]
LGTM. We like the approach though having the numbers initialised to something all the same would make us more comfortable as a preventative measure to ever accessing them uninitalised.

We'll merge this as is, but perhaps we can find values to set base and gpnvm_mask to that are clearly bad/wrong but which don't make this code do whacky stuff if ever triggered. Thinking about debug as much as anything.

**Comment by perigoso** [comment]
wait on the merge then, let me take another look

**Comment by perigoso** [comment]
added an additional check in the flash cmd for a non 0 eefc addr

**Review by dragonmux — APPROVED** [review]
Looks much better - happy to merge this once the build finishes as you've probably tested it too.

**Comment by perigoso** [comment]
ah bummer, i made one commit with the work email by mistake, oh well

**Pull request merged**
